doc/deploying_gcs_on_bare_metal_machines.md

```diff
@@ -0,0 +1,196 @@
+
+3. Stop firewalld on all machines<br/>
+   `$ systemctl stop firewalld`
+
+4. We will install kubernetes via the deploy-k8s script in the GCS repository. Clone GCS repository in master
+   node(kube1):<br/>
+  `$ git clone --recurse-submodules git@github.com:gluster/gcs.git`
+
+5. To install kubernetes we need create an inventory file mentioning the kubernetes nodes.<br/>
+
+   Template for GCS inventory file can be found under deploy directory in gcs repository:<br/>
+   `https://github.com/gluster/gcs/blob/master/deploy/examples/inventory-gcs-kubespray.example`
+
+   We will be deploying GCS after deploying kubernetes so we will have to provide GCS
+   nodes in the invntory file. Also, we will mention the devices that we want to use to create volumes.
+   Since we are using only 3 nodes we will provide devices for all the 3 nodes and mention
+   all the nodes in the gcs nodes.
+    ```
+    kube1 ansible_host=<node-ip> gcs_disks='["<device1>", "<device2>"]'
+    kube2 ansible_host=<node-ip> gcs_disks='["<device1>", "<device2>"]'
+    kube3 ansible_host=<node-ip> gcs_disks='["<device1>", "<device2>"]'
+
+    ## Hosts that will run etcd for the Kubernetes cluster
+    [etcd]
+    kube1
+    kube2
+    kube3
+
+    ## Hosts that will be kubernetes master nodes
+    [kube-master]
+    kube1
+
+    ## Hosts that will be kuberenetes nodes
+    [kube-node
+    kube1
+    kube2
+    kube3
+
+    ## The full kubernetes cluster
+    [k8s-cluster:children]
+    kube-master
+    kube-node
+
+    ## Hosts that will be used for GCS.
+    ## Systems grouped here need to define 'gcs_disks' as hostvars, which are the disks that will be used by GCS to 
+    provision storage.
+    [gcs-node]
+    kube1
+    kube2
+    kube3
+    ```
+
+
+6. To deploy kubernetes we need to excecute deploy-k8s.yml file using ansible and provide inventory file created in Step 5.<br/>
+   The deploy-k8s.yml file is present under deploy directory in gcs repository. <br />
+   `$ ansible-playbook -i <path to inventory file>/<name of inventory file>  <path to`
+     `deploy-k8s.yml file>/deploy-k8s.yml`
+
+7. Check whether all the nodes are in ready state.<br />
+   `$ kubectl get nodes`<br />
+
+    The status of all the nodes should be in READY state.
+
+> Note: In case deploy kubernetes fails then please check the troubleshooting section.
+
+### Deploying GCS on kubernetes cluster:
+
+#### Steps to install GCS:
+
+1. To deploy GCS on kubernetes cluster we will execute the deploy-gcs.yml file using ansible from master node(kube 1). The deploy-gcs.yml file is present under deploy directory.<br />
+   `$ ansible-playbook -i <path to inventory file>/<name of inventory file>  <path to deploy-gcs.yml file>/deploy-gcs.yml`.
+
+### Testing the deployment
+
+To test whether the deployment was successfull or not, you can run the following command on the master node.<br />
+`$ kubectl get pods -n gcs`<br />
+
+All the pods should be in `Running` state.
```

Comment:
`gluster-mixins` would be in `completed` state. So, above statement might confuse the user.
Also, how would I know what are "all the pods" that should ideally be running? 

I would suggest adding the screenshot of the above command in a successful deployment.

Reply:
I didn't add a screenshot because in future the number of pods or the name of pods would change and it might confuse user. Its better to provide a generic solution.

Reply:
@umangachapagain Why does gluster-mixins remains in completed state and not in running state?

Reply:
> @umangachapagain Why does gluster-mixins remains in completed state and not in running state?

Because, it has just one job i.e. to deploy dashboards and alerts. When this job completes, `gluster-mixin` will be in completed state which happens as soon as the whole deployment completes successfully.